File: 3423/3423.ts
function maximumDifferenceBetweenAdjace(root: TreeNode | null): number {
    if (!root) return 0;
    return 1 + Math.max(maximumDifferenceBetweenAdjace(root.left), maximumDifferenceBetweenAdjace(root.right));
}

File: 3423/3423.go
func maximumDifferenceBetweenAdjace(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumDifferenceBetweenAdjace(root.Left)
	r := maximumDifferenceBetweenAdjace(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}